src/tsdl_image.ml

```diff
@@ -29,7 +29,12 @@ end
    in the toplevel, see
    https://github.com/ocamllabs/ocaml-ctypes/issues/70 *)
 let foreign name typ =
-  foreign name typ ~from:Dl.(dlopen ~filename:"libSDL2_image-2.0.so.0"
+  let filename =
+    match Build_config.system with
+      | "macosx" -> "libSDL2_image-2.0.0.dylib"
+      | _ -> "libSDL2_image-2.0.so.0"
+  in
```

Comment:
the match case would be executed at each call to `foreign`, which adds some overhead. SDL functions are supposed to be executed very fast, so I'm afraid it could degrade performance. I suggest defining filename as a global variable when the library is loaded.

Reply:
Good call. Just pushed a factorization that should open the shared library only once.